source/reference/modules/deployer/index.rst

```diff
@@ -287,6 +287,54 @@ The Deployer out of the box provides the following processor beans:
 * **mailNotificationProcessor:** sends an email notification when there's a successful deployment with file changes or when
   a deployment failed.
 
+""""""""""""""""
+Thread Pool Size
+""""""""""""""""
+As the number of sites grows you may need more workers (threads) in the Deployer to service them. If you do not add more
+workers then you will see errors in scheduled tasks. Eventually, the system will get through the workload with the workers it
+has available, and the error will stop, but the presence of these errors on a regular basis indicates that you need
+more workers in the pool.
+
+To increase the number of workers, set the thread pool size of the task scheduler by setting the ``poolSize`` property
+in ``CRAFTER_HOME/bin/crafter-deployer/config/application.yaml`` as shown below:
+
+.. code-block:: yaml
+    :caption: *CRAFTER_HOME/bin/crafter-deployer/config/application.yaml*
+    :emphasize-lines: 3-5
+    :linenos:
+
+    deployer:
+      main:
+        taskScheduler:
+          # Thread pool size of the task scheduler
+          poolSize: 20
```

Comment:
I think it's also important to know that there's this pool too. The task scheduler pool is used for starting deployments on a schedule of every 10 secs, but the deployment pool is used to run all deployments

```
deployer:
  main:
    deployments:
      pool:
        # Thread pool core size
        size: 25
        # Thread pool max size
        max: 100
        # Thread pool queue size
        queue: 100
```